qiita_pet/handlers/api_proxy/tests/test_prep_template.py

```diff
@@ -6,28 +6,112 @@
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
 from unittest import TestCase, main
+from os import remove
+from os.path import join, exists
+from string import ascii_letters
+from random import choice
 
-from qiita_pet.handlers.api_proxy.prep_template import prep_template_get_req
+from qiita_core.qiita_settings import qiita_config
+from qiita_core.util import qiita_test_checker
+from qiita_db.ontology import Ontology
+from qiita_pet.handlers.api_proxy.prep_template import (
+    prep_template_summary_get_req, prep_template_post_req,
+    prep_template_put_req, prep_template_delete_req, prep_template_get_req,
+    prep_template_graph_get_req, prep_template_filepaths_get_req,
+    prep_ontology_get_req, _process_investigation_type)
 
 
+@qiita_test_checker()
 class TestPrepAPI(TestCase):
+    def setUp(self):
+        fp = join(qiita_config.base_data_dir, 'uploads', '1',
+                  'uploaded_file.txt')
+        if not exists(fp):
+            with open(fp, 'w') as f:
+                f.write('')
+
+        # Create test file to point update tests at
+        self.update_fp = join(qiita_config.base_data_dir, 'uploads', '1',
+                              'update.txt')
+        with open(self.update_fp, 'w') as f:
+            f.write("""sample_name\tnew_col\n1.SKD6.640190\tnew_value\n""")
+
+        def tear_down(self):
+            remove(self.update_fp)
+
+    def test_process_investigation_type(self):
+        obs = _process_investigation_type('Metagenomics', '', '')
+        self.assertEqual(obs, 'Metagenomics')
+
+    def test_process_investigation_type_user_term(self):
+        _process_investigation_type('Other', 'New Type', 'userterm')
+        obs = _process_investigation_type('Other', 'userterm', '')
+        self.assertEqual(obs, 'userterm')
+
+    def test_process_investigation_type_new_term(self):
+        randstr = ''.join([choice(ascii_letters) for x in range(30)])
+        obs = _process_investigation_type('Other', 'New Type', randstr)
+        self.assertEqual(obs, randstr)
+
+        # Make sure New Type added
+        ontology = Ontology(999999999)
+        self.assertIn(randstr, ontology.user_defined_terms)
+
+    def test_prep_ontology_get_req(self):
+        obs = prep_ontology_get_req()
+        exp = {'ENA': ['Cancer Genomics', 'Epigenetics', 'Exome Sequencing',
+                       'Forensic or Paleo-genomics', 'Gene Regulation Study',
+                       'Metagenomics', 'Pooled Clone Sequencing',
+                       'Population Genomics', 'RNASeq', 'Resequencing',
+                       'Synthetic Genomics', 'Transcriptome Analysis',
+                       'Whole Genome Sequencing', 'Other'],
+               'User': [],
+               'status': 'success',
+               'message': ''}
+        self.assertEqual(obs, exp)
+
     def test_prep_template_get_req(self):
```

Comment:
I think this test is incomplete. It is only testing that one of the samples is present. I'm not sure if that is extensive enough.